samples/src/main/java/okio/samples/Throttler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio.samples;
+
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.util.concurrent.TimeUnit;
+import okio.Buffer;
+import okio.ForwardingSink;
+import okio.ForwardingSource;
+import okio.Sink;
+import okio.Source;
+
+/**
+ * Enables throttling of Sources and Sinks. Attach to this throttler via {@link #source(Source)} and
+ * {@link #sink(Sink)} and set the desired throughput via {@link #bytesPerTimePeriod(long, long,
+ * TimeUnit)}. Multiple Sources and Sinks can be attached to a single Throttler and they will be
+ * throttled as a group, where their combined throughput will not exceed the desired throughput. The
+ * same Source or Sink can be attached to multiple Throttlers and its throughput will not exceed the
+ * desired throughput of any of the Throttlers.
+ *
+ * <p>To clear any throttling, use {@link #clear()} or specify a <code>byteCount</code> of <code>
+ * 0</code> when setting the throughput.
+ */
+public final class Throttler {
+  private static final int NANOS_PER_MILLI = 1000000;
+
+  private double limit = 0.0d;
+  private long last = System.nanoTime();
```

Comment:
should this be initialized to some sentinel value like -1, so the first read or write is never delayed?